datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -280,7 +283,8 @@ pub struct HashJoinExec {
     pub filter: Option<JoinFilter>,
     /// How the join is performed (`OUTER`, `INNER`, etc)
     pub join_type: JoinType,
-    /// The output schema for the join
+    /// The schema after join. Please be careful when using this schema,
+    /// if there is a projection, the schema isn't the same as the output schema.
     schema: SchemaRef,
```

Comment:
If it is easy to confuse, maybe change to `join_schema`?

Reply:
good idea.